src/process.h

```diff
@@ -67,6 +73,11 @@ int process_reap( uint32_t pid );
 uint32_t process_getpid();
 uint32_t process_getppid();
 
+int process_available_fd();
```

Comment:
These are all inconsistent with the `process` API.  Pass in the intended process to modify as a parameter, rather than relying on the `current` global pointer.  (Because we will need the ability to modify processes not currently running.)